### DISCUSSION TITLE
An OpenTelemetry scraping endpoint (or periodic push)

### DISCUSSION THREAD

**Discussion by thompson-tomo** [rabbitmq/rabbitmq-server#10251] [Other]
### Is your feature request related to a problem? Please describe.

I want all my metrics from all the different sources to visible in the 1 tool which can handle OTEL data 

### Describe the solution you'd like

I want an api to be exposed which provides metrics in OTEL format. This would enable depreciation of the prometheus endpoint.

Nice to have would be to have a builtin configurable push interval to eliminate polling

**Comment by michaelklishin** [comment]
This is still not specific enough to be an issue => moving to a discussion.

First of all, there are no plans to deprecate the Prometheus endpoint.

@ikavgo the topic of OpenTelemetry was investigated at least twice in the last couple of years. What was the conclusion, that we would like to do it eventually but without a lot of specifics?

**Reply by thompson-tomo** [reply]
See https://github.com/open-telemetry/opentelemetry-proto/blob/v1.0.0/opentelemetry/proto/collector/metrics/v1/metrics_service.proto for the grpc service describing the data to be transferred.

**Comment by michaelklishin** [comment]
This is not exactly the same topic but we have seen in https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1261, some messaging-related bits in OpenTelemetry are still immature. This has more to do with tracing than metrics per se.

But I assume the idea here is to use an OpenTelemetry format for scraping and nothing else.

**Reply by thompson-tomo** [reply]
Correct purely about exposing an endpoint to be scraped or pushing of metrics to an analysis tool. Transport occurs using a message compliant with OTEL allowing flexibility in tool used.
